new Beta 1.11

Raycoms opened the pull request:
It's been a while since we had the last beta release, this alpha runs for quite a while now quite smoothly.
I think it's time to release it for real.

I will let this pr open for a few days if there are no objections I will merge it

Review please

Asherslab committed:
Done (#1186)

xavierh committed:
Use the Buildertool as a unique Item in recipes to prevent conflicts

Raycoms committed:
Fix item missing when dman dumping (#1187) (#1197)

Raycoms committed:
Merge branch 'testing/1.11' into version/1.11

Raycoms committed:
Several bugfixes for Farmer and Baker

Raycoms committed:
port small-fixes to 1.11 (#1200)

Asherslab committed:
Makes the "Scepter"s and the Caliper look better when held (#1206)

Just makes the held item look a whole lot better

Raycoms committed:
Merged branch testing/1.11 into version/1.11

Raycoms committed:
Port/sapling selector + config rework (#1213)

* #1 of merging lumberjack sapling selector

* add 1.11.2 tinkers support

* adapt changes

* just stage

* stage

* first config changes

* Adapt config changes

* add missing last line for git

* fix orion annotations

* fix configuration

* register config changed event

* load configuration on preInit

* remove minecolonies custom gui fact

* rename lumberjack gui

* add fixes to 1.11 pr

* Fix some issues

Asherslab committed:
Commit =D (#1278)

fixes NullPointerErrors from console with /mc commands
fixes last seen of colony owner

pmardle committed:
Adds a recipe for the Double Fern (#1281)

Double fern recipe

Raycoms committed:
Port/manual assignment (#1277)

Port manual assignment
Nether fix
Creative builder inventory clicking

Asherslab committed:
Guide book(gigaherz) (#1243)

Add guidebook files to minecolonies and get it from the supplychest

marchermans committed:
[Clean Up] Initial parts of the Request system. (#1305)

* Adding the core API of the RequestSystem.
Adds:
  * Better control in the Factories.
  * IRequest and IRequestFactory
  * IRequestResolver, Provider and Factory.
  * IRequestManager, RequestState to control the Requests in a Colony.

* Working on the initial Core implementations for deliveries.

* Implementing the Chaining of a Delivery on a ItemStack request.

* Adding the core API of the RequestSystem.
Improving the Handling, of the SFC

Added:
  * Better control in the Factories.
  * IRequest and IRequestFactory
  * IRequestResolver, Provider and Factory.
  * IRequestManager, RequestState to control the Requests in a Colony.
  * Better handling of the SFC.
  * ReflectionUtils class that contains Utility methods with regards to reflection.
  * Tests that verify the functioning of the SFC
  * Tests that verify the functioning of the ReflectionsUtil Class.

* Adding the core API of the RequestSystem.
Improving the Handling, of the SFC

Added:
  * Better control in the Factories.
  * IRequest and IRequestFactory
  * IRequestResolver, Provider and Factory.
  * IRequestManager, RequestState to control the Requests in a Colony.
  * Better handling of the SFC.
  * ReflectionUtils class that contains Utility methods with regards to reflection.
  * Tests that verify the functioning of the SFC
  * Tests that verify the functioning of the ReflectionsUtil Class.

* Fixing a merge bug in the IFactoryController when a change was not Stage properly.

* Fixing the IColony javadoc.

* Run the formatter to make Sonar happy.

* Fixing a bug in the SFC.

 * Input types have no secondary types as it is not possible to pas a IToken into a Method parameter of type StandardToken.
 * Fixing several compatibility problems.

* Running the formatter again to improve the code.

* Fixing finals, formatting and Inspections.

* Part-One cleanup

* Fixy

* more codacy / sonar

* little bit more

* Added ToDo

* Suppression Constants

* slightly more suppression changes

Asherslab committed:
Barbarians 1.11.2 (#1324)

* Adds Barbarians to 1.11.2

Commit:
XP Changes port (#1333)

Bugfixes of xp gaining +
mending for worker tools

Asherslab committed:
Fixes barbarian commands to require OP (#1348)

* Player must be Op

Raycoms committed:
Fix several small bugs.

Asherslab committed:
Donie (#1379)

Raycoms committed:
Port/racks to 1.11 (#1367)

Port racks to 1.11

Asherslab committed:
Done (#1366)

Fix Mc update commands in 1.11.2 as well

MineShadow7 committed:
First (#1377)

Raycoms committed:
Add journeymap support + npe fix (#1383) (#1391)

Raycoms committed:
Adding a visual Block for Waypoint Placement. (#1387)

Asherslab committed:
Fix Null spawnpoints of barbarians in 1.11

Raycoms committed:
fix blockParticleEffect message (#1396)

fix blockParticleMessage and waypoint string

Raycoms committed:
Port to 1.12 (#1395)

This pr, ports everything which we haven't ported yet from 1.11 to 1.12.
Including, waypoint, racks, timberframe

Raycoms committed:
Feature/citizen info sign (#1386)

Fixes entity rendering of npcs.
Adds informal things onto a sign which will show up when getting close.
Fixes problem with name rendering.
Fixes problem with dman not leaving enough items behind
Fixes problem with dman not leaving dirt at the miner

Raycoms committed:
Revert "Port to 1.12" (#1401)

* Revert "Feature/citizen info sign (#1386)"

This reverts commit 31c9a1e52e45fbf0eb5d2f12892741faf7e73c49.

* Revert "Port to 1.12 (#1395)"

This reverts commit be0275cec05d5289a8d9c5824c4956e7ecb70fb3.

Raycoms committed:
Feature/gui (#1398)

* Add textures improve waypoint schematics

* Redesign several of the townhall pages

+ bugfixing

* finish GUI redsign

* Improve the style

* final adjustments

* add corners as well

Raycoms committed:
Port/bug fixes (#1407)

Improve standard spawn times.
Avoid NPE by Message.
let the builder find easier building locations
Let workers pick up items a block above or under them.

Raycoms committed:
Protect villages from being overbuilt (#1399)

* protect villages from being overbuilt

only on config option

* Drop wrong entities during the build

* fix itemFrames

* fix codacity

Raycoms committed:
let citizens sleep close to their beds (#1423) (#1431)

* let citizens sleep close to their beds (#1423)

Raycoms committed:
Feature/increase storage (#1416)

Offer to increase the warehouse in the GUI (from level 5 on)

Raycoms committed:
Feature/1362 improve rendering (#1420)

Fixes blockStates when rendering
Preloaded models only once when rendering (performance improvement)

Raycoms committed:
fix sign rendering (#1435)

Raycoms committed:
Port1.11/barracks (#1440)

* merge Barracks to 1.11

* fix merge problems

Raycoms committed:
Merge branch 'testing/1.11' into version/1.11

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/Colony.java:
```diff
@@ -180,28 +191,31 @@
-    private int maxCitizens  = Configurations.maxCitizens;
+    private int maxCitizens  = Configurations.gameplay.maxCitizens;
 
     private double overallHappiness = 5;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 5 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/Colony.java:
```diff
@@ -1073,14 +1094,19 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
             if (townHall != null && citizens.size() < maxCitizens)
             {
-                int respawnInterval = Configurations.citizenRespawnInterval * 20;
+                int respawnInterval = Configurations.gameplay.citizenRespawnInterval * 20;
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 20 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/Colony.java:
```diff
@@ -1073,14 +1094,19 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
             {
-                int respawnInterval = Configurations.citizenRespawnInterval * 20;
+                int respawnInterval = Configurations.gameplay.citizenRespawnInterval * 20;
                 respawnInterval -= (60 * townHall.getBuildingLevel());
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 60 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/Colony.java:
```diff
@@ -1197,7 +1223,7 @@ public World getWorld()
     {
-        final int distanceFromCenter = Configurations.workingRangeTownHall + 48 /* 3 chunks */ + 15 /* round up a chunk */;
+        final int distanceFromCenter = Configurations.gameplay.workingRangeTownHall + 48 /* 3 chunks */ + 15 /* round up a chunk */;
         for (int x = -distanceFromCenter; x <= distanceFromCenter; x += 16)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/Colony.java:
```diff
@@ -1197,7 +1223,7 @@ public World getWorld()
+        final int distanceFromCenter = Configurations.gameplay.workingRangeTownHall + 48 /* 3 chunks */ + 15 /* round up a chunk */;
         for (int x = -distanceFromCenter; x <= distanceFromCenter; x += 16)
         {
             for (int z = -distanceFromCenter; z <= distanceFromCenter; z += 16)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 16 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/entity/EntityCitizen.java:
```diff
@@ -601,11 +674,12 @@ public void gatherXp()
      */
     public void addExperience(final double xp)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 25 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/locations/EntityLocation.java:
```diff
@@ -0,0 +1,179 @@
+        return !(entity == null || entity.get() == null) && location.getDimension() == getDimension();
+    }
+
+    @SuppressWarnings("squid:S2972")
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S2972' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/locations/EntityLocation.java:
```diff
@@ -0,0 +1,179 @@
+
+        @NotNull
+        @Override
+        @SuppressWarnings("squid:LeftCurlyBraceStartLineCheck")
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:LeftCurlyBraceStartLineCheck' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/locations/EntityLocation.java:
```diff
@@ -0,0 +1,179 @@
+
+        @NotNull
+        @Override
+        @SuppressWarnings("squid:LeftCurlyBraceStartLineCheck")
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:LeftCurlyBraceStartLineCheck' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java:
```diff
@@ -28,26 +36,40 @@
 public class BuildingLumberjack extends AbstractBuildingWorker
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override the "equals" method in this class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2160)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/BuildingLumberjack.java:
```diff
@@ -160,8 +249,12 @@ public String getJobName()
     public static class View extends AbstractBuildingWorker.View
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 70 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/locations/StaticLocation.java:
```diff
@@ -0,0 +1,168 @@
+
+        @NotNull
+        @Override
+        @SuppressWarnings("squid:LeftCurlyBraceStartLineCheck")
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:LeftCurlyBraceStartLineCheck' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/locations/StaticLocation.java:
```diff
@@ -0,0 +1,168 @@
+    /**
+     * Internal factory class.
+     */
+    @SuppressWarnings("squid:S2972")
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S2972' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/locations/StaticLocation.java:
```diff
@@ -0,0 +1,168 @@
+
+        @NotNull
+        @Override
+        @SuppressWarnings("squid:LeftCurlyBraceStartLineCheck")
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:LeftCurlyBraceStartLineCheck' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/AbstractRequest.java:
```diff
@@ -0,0 +1,407 @@
+    private R      result;
+    @Nullable
+    private IToken parent;
+    @SuppressWarnings("squid:S1170")
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S1170' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

MinecoloniesSonar commented on src/main/java/com/minecolonies/blockout/controls/TextField.java:
```diff
@@ -367,27 +367,27 @@ else if (cursorBeforeEnd && shadow)
         }
 
         //  Should we draw the cursor this frame?
         if (isFocus() && cursorVisible && (cursorBlinkCounter / 6 % 2 == 0))
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Assign this magic number 6 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java:
```diff
@@ -51,9 +50,10 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
     @Override
     public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 9, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java:
```diff
@@ -0,0 +1,169 @@
+    @Override
+    public void onResolvingOverruled(@NotNull final IRequestManager manager, @NotNull final IRequest<ItemStack> request)
+    {
+        //TODO Release the stack from the warehouse.
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java:
```diff
@@ -0,0 +1,169 @@
+        //TODO Release the stack from the warehouse.
+    }
+
+    @SuppressWarnings("squid:S2972")
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:S2972' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java:
```diff
@@ -0,0 +1,169 @@
+    @Override
+    public void onRequestComplete(@NotNull final IToken token)
+    {
+        //TODO Release the DMan that did the job.
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java:
```diff
@@ -0,0 +1,169 @@
+    @Override
+    public IRequest onParentCancelled(@NotNull final IRequestManager manager, @NotNull final IRequest<ItemStack> request)
+    {
+        //TODO Release the stack from the warehouse.
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/resolvers/WarehouseRequestResolver.java:
```diff
@@ -0,0 +1,169 @@
+
+    @Nullable
+    @Override
+    @SuppressWarnings("squid:LeftCurlyBraceStartLineCheck")
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Suppressing the 'squid:LeftCurlyBraceStartLineCheck' warning is not allowed [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1309)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/citizencommands/AbstractCitizensCommands.java:
```diff
@@ -44,24 +44,24 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
     @Override
     public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequestFactories.java:
```diff
@@ -0,0 +1,283 @@
+            }
+
+            @SuppressWarnings(Suppression.LEFT_CURLY_BRACE)
+            final StandardRequests.ItemStackRequest request = controller.getNewInstance(requested, new TypeToken<StandardRequests.ItemStackRequest>() {}, token, state);
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequestFactories.java:
```diff
@@ -0,0 +1,283 @@
+    }
+
+    @SuppressWarnings(Suppression.BIG_CLASS)
+    public static final class DeliveryFactory implements IRequestFactory<Delivery, StandardRequests.DeliveryRequest>
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 80 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequestFactories.java:
```diff
@@ -0,0 +1,283 @@
+        @SuppressWarnings(Suppression.LEFT_CURLY_BRACE)
+        public TypeToken<Delivery> getFactoryInputType()
+        {
+            return new TypeToken<Delivery>() {};
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequestFactories.java:
```diff
@@ -0,0 +1,283 @@
+        @SuppressWarnings(Suppression.LEFT_CURLY_BRACE)
+        public TypeToken<StandardRequests.ItemStackRequest> getFactoryOutputType()
+        {
+            return new TypeToken<StandardRequests.ItemStackRequest>() {};
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequestFactories.java:
```diff
@@ -0,0 +1,283 @@
+            }
+
+            @SuppressWarnings(Suppression.LEFT_CURLY_BRACE)
+            final StandardRequests.DeliveryRequest request = controller.getNewInstance(requested, new TypeToken<StandardRequests.DeliveryRequest>() {}, token, state);
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequestFactories.java:
```diff
@@ -0,0 +1,283 @@
+    }
+
+    @SuppressWarnings(Suppression.BIG_CLASS)
+    public static final class ItemStackFactory implements IRequestFactory<ItemStack, StandardRequests.ItemStackRequest>
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 80 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequestFactories.java:
```diff
@@ -0,0 +1,283 @@
+        @SuppressWarnings(Suppression.LEFT_CURLY_BRACE)
+        public TypeToken<StandardRequests.DeliveryRequest> getFactoryOutputType()
+        {
+            return new TypeToken<StandardRequests.DeliveryRequest>() {};
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/requestsystem/requests/StandardRequestFactories.java:
```diff
@@ -0,0 +1,283 @@
+        @SuppressWarnings(Suppression.LEFT_CURLY_BRACE)
+        public TypeToken<ItemStack> getFactoryInputType()
+        {
+            return new TypeToken<ItemStack>() {};
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this left curly brace to the beginning of next line of code. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ALeftCurlyBraceStartLineCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java:
```diff
@@ -121,6 +123,9 @@
         addMapping("Fisherman", BuildingFisherman.class, BuildingFisherman.View.class, BlockHutFisherman.class);
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this reference to "BuildingFisherman". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this reference to "BuildingFisherman". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java:
```diff
@@ -121,6 +123,9 @@
         addMapping("Fisherman", BuildingFisherman.class, BuildingFisherman.View.class, BlockHutFisherman.class);
         addMapping("GuardTower", BuildingGuardTower.class, BuildingGuardTower.View.class, BlockHutGuardTower.class);
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this reference to "BuildingGuardTower". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this reference to "BuildingGuardTower". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java:
```diff
@@ -121,6 +123,9 @@
         addMapping("Fisherman", BuildingFisherman.class, BuildingFisherman.View.class, BlockHutFisherman.class);
         addMapping("GuardTower", BuildingGuardTower.class, BuildingGuardTower.View.class, BlockHutGuardTower.class);
         addMapping("WareHouse", BuildingWareHouse.class, BuildingWareHouse.View.class, BlockHutWareHouse.class);
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this reference to "BuildingWareHouse". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this reference to "BuildingWareHouse". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java:
```diff
@@ -121,6 +123,9 @@
         addMapping("Fisherman", BuildingFisherman.class, BuildingFisherman.View.class, BlockHutFisherman.class);
         addMapping("GuardTower", BuildingGuardTower.class, BuildingGuardTower.View.class, BlockHutGuardTower.class);
         addMapping("WareHouse", BuildingWareHouse.class, BuildingWareHouse.View.class, BlockHutWareHouse.class);
+        addMapping("Barracks", BuildingBarracks.class, BuildingBarracks.View.class, BlockHutBarracks.class);
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this reference to "BuildingBarracks". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this reference to "BuildingBarracks". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuilding.java:
```diff
@@ -121,6 +123,9 @@
         addMapping("GuardTower", BuildingGuardTower.class, BuildingGuardTower.View.class, BlockHutGuardTower.class);
         addMapping("WareHouse", BuildingWareHouse.class, BuildingWareHouse.View.class, BlockHutWareHouse.class);
+        addMapping("Barracks", BuildingBarracks.class, BuildingBarracks.View.class, BlockHutBarracks.class);
+        addMapping("BarracksTower", BuildingBarracksTower.class, BuildingBarracksTower.View.class, BlockHutBarracksTower.class);
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this reference to "BuildingBarracksTower". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this reference to "BuildingBarracksTower". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2390)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java:
```diff
@@ -535,15 +576,15 @@ private int findSaplingSlot()
     }
 
     //todo: we need to use a different way to get Metadata
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/lumberjack/EntityAIWorkLumberjack.java:
```diff
@@ -557,15 +598,22 @@ private void placeSaplings(final int saplingSlot, @NotNull final ItemStack stack
     }
 
     //todo: we need to use a different way to get Metadata
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java:
```diff
@@ -0,0 +1,98 @@
+    @SuppressWarnings(DEPRECATION)
+    @Override
+    @Deprecated
+    public boolean isFullBlock(final IBlockState state)
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java:
```diff
@@ -0,0 +1,98 @@
+     *
+     * @return true
+     */
+    //todo: remove once we no longer need to support this
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockWaypoint.java:
```diff
@@ -0,0 +1,98 @@
+    @SuppressWarnings(DEPRECATION)
+    @Override
+    @Deprecated
+    public boolean isOpaqueCube(final IBlockState state)
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java:
```diff
@@ -0,0 +1,666 @@
+        /**
+         * The list of manual patrol targets.
+         */
+        public List<BlockPos> patrolTargets = new ArrayList<>();
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make patrolTargets a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java:
```diff
@@ -0,0 +1,666 @@
+        /**
+         * The task of the guard, following the Task enum.
+         */
+        public Task task = Task.GUARD;
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make task a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java:
```diff
@@ -0,0 +1,666 @@
+        /**
+         * Assign the job manually, knight or ranger.
+         */
+        public boolean assignManually = false;
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make assignManually a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java:
```diff
@@ -0,0 +1,666 @@
+        /**
+         * Position the guard should guard at.
+         */
+        public BlockPos guardPos = this.getID();
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make guardPos a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java:
```diff
@@ -0,0 +1,666 @@
+        /**
+         * Retrieve the guard on low health.
+         */
+        public boolean retrieveOnLowHealth = false;
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make retrieveOnLowHealth a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java:
```diff
@@ -0,0 +1,666 @@
+        /**
+         * The job of the guard, following the GuarJob enum.
+         */
+        public GuardJob job = null;
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make job a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java:
```diff
@@ -0,0 +1,666 @@
+        /**
+         * Patrol manually or automatically.
+         */
+        public boolean patrolManually = false;
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Make patrolManually a static final constant or non-public and provide accessors if needed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AClassVariableVisibilityCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java:
```diff
@@ -0,0 +1,666 @@
+    /**
+     * The client view for the baker building.
+     */
+    public static class View extends AbstractBuildingWorker.View
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 58 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingGuards.java:
```diff
@@ -0,0 +1,666 @@
+/**
+ * Building class of the guard tower.
+ */
+public abstract class AbstractBuildingGuards extends AbstractBuildingWorker
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override the "equals" method in this class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2160)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java:
```diff
@@ -76,34 +109,25 @@ public CitizenData getWorker()
     public void setWorker(final CitizenData citizen)
     {
-        if (worker == citizen)
+        if (workers.contains(citizen))
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java:
```diff
@@ -272,29 +326,22 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
     /**
      * AbstractBuildingWorker View for clients.
      */
     public static class View extends AbstractBuildingHut.View
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 52 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java:
```diff
@@ -182,28 +235,28 @@ public void onDestroyed()
     public boolean isWorker(final CitizenData citizen)
     {
-        return citizen == worker;
+        return workers.contains(citizen);
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "contains()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java:
```diff
@@ -219,27 +272,28 @@ public void removeCitizen(final CitizenData citizen)
         {
-            setWorker(null);
+            citizen.setWorkBuilding(null);
+            workers.remove(citizen);
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') This call to "remove()" may be a performance hot spot if the collection is large. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2250)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java:
```diff
@@ -257,8 +311,8 @@ public void onWorldTick(@NotNull final TickEvent.WorldTickEvent event)
 
         // If we have no active worker, grab one from the Colony
         // TODO Maybe the Colony should assign jobs out, instead?
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Complete the task associated to this TODO comment. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1135)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java:
```diff
@@ -238,6 +315,15 @@ public boolean hasResident(final CitizenData citizen)
+    /**
      * The view of the citizen hut.
      */
     public static class View extends AbstractBuildingHut.View
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 38 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/BuildingHome.java:
```diff
@@ -6,25 +6,60 @@
 /**
  * The class of the citizen hut.
  */
 public class BuildingHome extends AbstractBuildingHut
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override the "equals" method in this class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2160)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/deliveryman/EntityAIWorkDeliveryman.java:
```diff
@@ -503,64 +519,62 @@ else if (itemsToDeliver.isEmpty())
     private AIState gatherItems(@NotNull final AbstractBuilding buildingToDeliver, @NotNull final BlockPos position)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIRangeGuard.java:
```diff
@@ -179,24 +184,38 @@ protected AIState searchTarget()
      */
     protected AIState huntDown()
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/builder/ConstructionTapeHelper.java:
```diff
@@ -56,7 +56,7 @@ public static void placeConstructionTape(@NotNull final AbstractBuilding buildin
 
     public static void placeConstructionTape(@NotNull final WorkOrderBuildDecoration workOrder, @NotNull final World world)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this method to reduce its Cognitive Complexity from 21 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/guard/EntityAIMeleeGuard.java:
```diff
@@ -124,27 +130,33 @@ protected AIState searchTarget()
      */
     protected AIState huntDown()
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/BuildingBarracksTower.java:
```diff
@@ -0,0 +1,148 @@
+/**
+ * Building class of the BarracksTower.
+ */
+public class BuildingBarracksTower extends AbstractBuildingGuards
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Override the "equals" method in this class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2160)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java:
```diff
@@ -0,0 +1,441 @@
+     */
+    @Override
+    @Deprecated
+    public boolean isOpaqueCube(final IBlockState state)
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java:
```diff
@@ -0,0 +1,441 @@
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState withRotation(@NotNull final IBlockState state, final Rotation rot)
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java:
```diff
@@ -0,0 +1,441 @@
+     * @deprecated but we still need this because there is nothing better.
+     */
+    @Override
+    public IBlockState getActualState(final IBlockState state, final IBlockAccess worldIn, final BlockPos pos)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add the missing @Deprecated annotation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMissingDeprecatedCheck)
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java:
```diff
@@ -0,0 +1,441 @@
+                }
+            }
+        }
+        super.neighborChanged(state, worldIn, pos, blockIn, fromPos);    }
```
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this closing curly brace to the next line. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3ARightCurlyBraceStartLineCheck)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java:
```diff
@@ -0,0 +1,441 @@
+     */
+    @Override
+    @Deprecated
+    public boolean isFullCube(final IBlockState state)
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java:
```diff
@@ -0,0 +1,441 @@
+     */
+    @Deprecated
+    @Override
+    public IBlockState getStateFromMeta(final int meta)
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java:
```diff
@@ -0,0 +1,441 @@
+    @NotNull
+    @Override
+    @Deprecated
+    public IBlockState withMirror(@NotNull final IBlockState state, final Mirror mirrorIn)
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java:
```diff
@@ -0,0 +1,441 @@
+        return BlockRenderLayer.SOLID;
+    }
+
+    public enum EnumType implements IStringSerializable
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce this class from 51 to the maximum allowed 30 or externalize it in a public class. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2972)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/blocks/BlockMinecoloniesRack.java:
```diff
@@ -0,0 +1,441 @@
+     */
+    @Override
+    @Deprecated
+    public boolean isFullBlock(final IBlockState state)
```
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1133)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/commands/generalcommands/RandomTeleportCommand.java:
```diff
@@ -155,14 +160,14 @@ private static void teleportPlayer(final ICommandSender sender, final EntityPlay
                 else
                 {
-                    sender.getCommandSenderEntity().sendMessage(new TextComponentString("Please wait at least " + Configurations.teleportBuffer + " seconds to teleport again"));
+                    sender.getCommandSenderEntity().sendMessage(new TextComponentString("Please wait at least " + Configurations.gameplay.teleportBuffer + " seconds to teleport again"));
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Split this 186 characters long line (which is greater than 180 authorized). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00103)

Asherslab commented:
i'll go through all the sonar and codacy errors and try to fix as many as possible soonish

Raycoms committed:
Feature/copy paste (#1436) (#1456)

It adds the possibility to paste messages directly into the world without requiring the builder.
For that, the player has to be in creative mode.
For now, it pastes it into the world with the correct building level + using the solid Substitution blocks.
In the future, we can add a second mode which turns of the solid substitution blocks.
It also fixes some bugs with the barracks.
And it fixes a bug where people lost their job after loading the barracks world.

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/colony/buildings/AbstractBuildingWorker.java:
```diff
@@ -128,50 +152,92 @@ public boolean neededForWorker(@Nullable final ItemStack stack)
+                    }
+                }
+            }
+            catch(final Exception e)
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Either log or rethrow this exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1166)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java:
```diff
@@ -125,14 +129,12 @@ private AIState startWorkingAtOwnBuilding()
     private AIState prepareForFarming()
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

MinecoloniesSonar commented on src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer/EntityAIWorkFarmer.java:
```diff
@@ -499,112 +367,168 @@ private boolean shouldHarvest(@NotNull final BlockPos position)
      * Checks the block above to see if it is a plant, if so, breaks it. Then tills.
      */
-    private AIState initialize()
+    private AIState workAtField()
```
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 8, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

MinecoloniesSonar commented:
SonarQube analysis reported 393 issues
* ![BLOCKER][BLOCKER] 3 blocker
* ![MAJOR][MAJOR] 235 major
* ![MINOR][MINOR] 108 minor
* ![INFO][INFO] 47 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [Structure.java#L137](https://github.com/Minecolonies/minecolonies/blob/5f36983d91b8256884746b3a453ca7b14104f0ec/src/main/java/com/minecolonies/structures/helpers/Structure.java#L137): Close this "InputStream" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2095)
1. ![BLOCKER][BLOCKER] [Structure.java#L247](https://github.com/Minecolonies/minecolonies/blob/5f36983d91b8256884746b3a453ca7b14104f0ec/src/main/java/com/minecolonies/structures/helpers/Structure.java#L247): Close this "InputStream" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2095)
1. ![BLOCKER][BLOCKER] [Structure.java#L260](https://github.com/Minecolonies/minecolonies/blob/5f36983d91b8256884746b3a453ca7b14104f0ec/src/main/java/com/minecolonies/structures/helpers/Structure.java#L260): Close this "InputStream" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2095)
1. ![MAJOR][MAJOR] [Pane.java#L19](https://github.com/Minecolonies/minecolonies/blob/5f36983d91b8256884746b3a453ca7b14104f0ec/src/main/java/com/minecolonies/blockout/Pane.java#L19): class "Pane" has 47 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ButtonImage.java#L13](https://github.com/Minecolonies/minecolonies/blob/5f36983d91b8256884746b3a453ca7b14104f0ec/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L13): class "ButtonImage" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ButtonImage.java#L563](https://github.com/Minecolonies/minecolonies/blob/5f36983d91b8256884746b3a453ca7b14104f0ec/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L563): Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)
1. ![MAJOR][MAJOR] [TextField.java#L21](https://github.com/Minecolonies/minecolonies/blob/5f36983d91b8256884746b3a453ca7b14104f0ec/src/main/java/com/minecolonies/blockout/controls/TextField.java#L21): class "TextField" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [TextField.java#L28](https://github.com/Minecolonies/minecolonies/blob/5f36983d91b8256884746b3a453ca7b14104f0ec/src/main/java/com/minecolonies/blockout/controls/TextField.java#L28): Assign this magic number 32 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [TextField.java#L325](https://github.com/Minecolonies/minecolonies/blob/5f36983d91b8256884746b3a453ca7b14104f0ec/src/main/java/com/minecolonies/blockout/controls/TextField.java#L325): The Cyclomatic Complexity of this method "drawSelf" is 21 which is greater than 15 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [TextField.java#L325](https://github.com/Minecolonies/minecolonies/blob/5f36983d91b8256884746b3a453ca7b14104f0ec/src/main/java/com/minecolonies/blockout/controls/TextField.java#L325): Refactor this method to reduce its Cognitive Complexity from 24 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

Raycoms committed:
Port1.11/fix/rendering (#1474)

* fix rendering of racks with the build tool

* improve

* fix merge problems

MinecoloniesSonar commented:
SonarQube analysis reported 393 issues
* ![BLOCKER][BLOCKER] 3 blocker
* ![MAJOR][MAJOR] 235 major
* ![MINOR][MINOR] 108 minor
* ![INFO][INFO] 47 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![BLOCKER][BLOCKER] [Structure.java#L137](https://github.com/Minecolonies/minecolonies/blob/d64985c2fd39d206e36fd25c1b4113fb1a819001/src/main/java/com/minecolonies/structures/helpers/Structure.java#L137): Close this "InputStream" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2095)
1. ![BLOCKER][BLOCKER] [Structure.java#L247](https://github.com/Minecolonies/minecolonies/blob/d64985c2fd39d206e36fd25c1b4113fb1a819001/src/main/java/com/minecolonies/structures/helpers/Structure.java#L247): Close this "InputStream" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2095)
1. ![BLOCKER][BLOCKER] [Structure.java#L260](https://github.com/Minecolonies/minecolonies/blob/d64985c2fd39d206e36fd25c1b4113fb1a819001/src/main/java/com/minecolonies/structures/helpers/Structure.java#L260): Close this "InputStream" in a "finally" clause. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2095)
1. ![MAJOR][MAJOR] [Pane.java#L19](https://github.com/Minecolonies/minecolonies/blob/d64985c2fd39d206e36fd25c1b4113fb1a819001/src/main/java/com/minecolonies/blockout/Pane.java#L19): class "Pane" has 47 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ButtonImage.java#L13](https://github.com/Minecolonies/minecolonies/blob/d64985c2fd39d206e36fd25c1b4113fb1a819001/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L13): class "ButtonImage" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [ButtonImage.java#L563](https://github.com/Minecolonies/minecolonies/blob/d64985c2fd39d206e36fd25c1b4113fb1a819001/src/main/java/com/minecolonies/blockout/controls/ButtonImage.java#L563): Extract this nested ternary operation into an independent statement. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3358)
1. ![MAJOR][MAJOR] [TextField.java#L21](https://github.com/Minecolonies/minecolonies/blob/d64985c2fd39d206e36fd25c1b4113fb1a819001/src/main/java/com/minecolonies/blockout/controls/TextField.java#L21): class "TextField" has 38 methods, which is greater than the 35 authorized. Split it into smaller classes. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [TextField.java#L28](https://github.com/Minecolonies/minecolonies/blob/d64985c2fd39d206e36fd25c1b4113fb1a819001/src/main/java/com/minecolonies/blockout/controls/TextField.java#L28): Assign this magic number 32 to a well-named constant, and use the constant instead. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [TextField.java#L325](https://github.com/Minecolonies/minecolonies/blob/d64985c2fd39d206e36fd25c1b4113fb1a819001/src/main/java/com/minecolonies/blockout/controls/TextField.java#L325): The Cyclomatic Complexity of this method "drawSelf" is 21 which is greater than 15 authorized. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)
1. ![MAJOR][MAJOR] [TextField.java#L325](https://github.com/Minecolonies/minecolonies/blob/d64985c2fd39d206e36fd25c1b4113fb1a819001/src/main/java/com/minecolonies/blockout/controls/TextField.java#L325): Refactor this method to reduce its Cognitive Complexity from 24 to the 20 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)

[BLOCKER]: https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'
